docker/Dockerfile.ci_cpu

```diff
@@ -13,6 +13,10 @@ RUN bash /install/ubuntu_install_core.sh
 COPY install/ubuntu_install_build.sh /install/ubuntu_install_build.sh
 RUN bash /install/ubuntu_install_build.sh
 
+# tcmalloc
+RUN apt-get install -y libgoogle-perftools4
+ENV LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libtcmalloc.so.4:$LD_PRELOAD
```

Comment:
Once prepend the tcmalloc to `LD_PRELOAD`, will `dgllib.so` always use tcmalloc instead glic for memory allocation automatically?

Reply:
do we need to link tcmalloc when building dgllib.so?

Reply:
Yes, `tcmalloc` will be used for all glibc functions for PyTorch and DGL(and other).
Unfortunately, linkage with tcmalloc doesn't work properly, that's why we propose to use `PRELOAD`

Reply:
I see. But updating the docker image(we need to build new docker image with the changed dockerfile) benefit CI tests only.If `tcmalloc` proves to always outperform `glibc`, we'd better mention such performance improvement in user guide? Then user could try with `tcmalloc` if they want. Or if DGL tries to force user to always use tcmalloc, DGL lib should be built/linked with it directly?

Reply:
Good idea, added to the user guide.
Unfortunately it is seems not possible to build DGL with tcmalloc since pytorch is not built with it. Mismatch in memory allocators causes crashes